test/integration/security/groups_users_permissions_sanity.js

```diff
@@ -1,410 +1,514 @@
+            expect(result.custom_data.customNumber == anotherSubGroup.custom_data.customNumber).to.be(
+              true
+            );
+            callback();
+          }
+        );
       });
-    });
-
-    it('should create a sub group', function (callback) {
-      testServices.security.users.upsertGroup(subGroup, {
-        parent: addedGroup
-      }, function (e, result) {
-
-        if (e) return callback(e);
-
-        expect(result.name == subGroup.name).to.be(true);
-        expect(result.custom_data.customString == subGroup.custom_data.customString).to.be(true);
-        expect(result.custom_data.customNumber == subGroup.custom_data.customNumber).to.be(true);
-
-        addedSubGroup = result;
-        callback();
 
+      it('should fail to create a group as it already exists', function(callback) {
+        testServices.security.users.upsertGroup(
+          testGroup,
+          {
+            overwrite: false
+          },
+          function(e) {
+            if (
+              e &&
+              e.toString() ==
+                'Error: validation failure: group by the name ' + testGroup.name + ' already exists'
+            )
+              return callback();
+
+            callback(new Error('group was created or lookup failed', e));
+          }
+        );
       });
-    });
-
-    it('should fail to create a group as it already exists', function (callback) {
-      testServices.security.users.upsertGroup(testGroup, {
-        overwrite: false
-      }, function (e, result) {
-
-        if (e && e.toString() == 'Error: validation failure: group by the name ' + testGroup.name + ' already exists')
-          return callback();
 
-        callback(new Error('group was created or lookup failed', e));
+      it('should get groups by group name', function(callback) {
+        testServices.security.users.listGroups('TEST*', function(e, results) {
+          if (e) return callback(e);
 
+          expect(results.length).to.be(3);
+          callback();
+        });
       });
-    });
-
-    it('should get groups by group name', function (callback) {
 
-      testServices.security.users.listGroups('TEST*', function (e, results) {
-
-        if (e) return callback(e);
+      it('gets a specific group', function(callback) {
+        testServices.security.users.getGroup(testGroup.name, function(e, group) {
+          if (e) return callback(e);
 
-        expect(results.length).to.be(2);
-        callback();
+          expect(group.name).to.be(testGroup.name);
 
+          callback();
+        });
       });
-    });
-
-    it('gets a specific group', function (callback) {
-
-      testServices.security.users.getGroup(testGroup.name, function (e, group) {
-
-        if (e) return callback(e);
-
-        expect(group.name).to.be(testGroup.name);
-
-        callback();
 
+      it('should get groups matching special criteria', function(callback) {
+        testServices.security.users.listGroups('TEST*', {
+          criteria:{

```

Comment:
Will options also work?
``` 
testServices.security.users.listGroups('TEST*', {
  criteria:{
    "custom_data.cadre":{$gt:0}
  },
  options:{
     limit:1
  }
}, callback); 
```

Reply:
I think we should get this out, then I will extend tests to be sure